pages/docs/examples/ai-agents-and-rag.mdx

```diff
@@ -0,0 +1,156 @@
+import { ResourceGrid, Resource, ExampleGrid } from 'src/shared/Docs/Resources';
+import { Example } from 'src/shared/Docs/Examples';
+
+import { BookOpenIcon, VideoCameraIcon, MicrophoneIcon } from "@heroicons/react/24/outline";
+
+export const description = "Learn how to use Inngest for AI automated workflows, AI agents, and RAG."
+
+# AI Agents and RAG
+
+Inngest offers tools to support the development of AI-powered applications. Whether you're building AI agents, automating tasks, or orchestrating and managing AI workflows, Inngest provides features that accommodate various needs and requirements, such as concurrency, debouncing, or throttling (see ["Related Concepts"](#related-concepts)).
+
+## Quick Snippet
+
+Below is an example of a RAG workflow (from the ["AI in production"](/blog/ai-in-production-managing-capacity-with-flow-control) blog post), which uses [Inngest steps](/docs/learn/inngest-steps) to guarantee automatic retries on failure.
+
+```typescript {{ title: "./inngest/functions.ts" }}
```

Comment:
I wonder if maybe [this code](https://github.com/inngest/inngest-demo-app/blob/7fa5e43bd6b3f7d767200fd5c449e2dc165ca033/inngest/ai.ts#L58-L124) would be better here? WDYT?

Reply:
I agree that it is simpler to read!